Generic_Environment.php

```diff
@@ -207,4 +209,87 @@ public function is_advanced_cache_add_in() {
 		return ( ( $script_data = @file_get_contents( W3TC_ADDIN_FILE_ADVANCED_CACHE ) )
 			&& strstr( $script_data, 'PgCache_ContentGrabber' ) !== false );
 	}
+
+	/**
+	 * Enables crawling blocker for cache directory
+	 *
+	 * @throws Util_WpFile_FilesystemOperationException with S/FTP form if it can't get the required filesystem credentials
+	 */
+	private function robots_add_directive() {
+		$robots_path = Util_Environment::site_path() . 'robots.txt';
+
+		$robots_content = @file_get_contents( $robots_path );
+		if ( $robots_content === false )
```

Comment:
WPCS: Use Yoda conditions and braces.
```
if ( false === $robots_content ) {
```

Did you mean to return if a `robots.txt` file was not found?  I would think that you would set the variable to an empty string.